metadata-io/src/main/java/com/linkedin/metadata/system_info/SystemInfoService.java

```diff
@@ -0,0 +1,142 @@
+  private final ExecutorService executorService = Executors.newFixedThreadPool(10);
+
+  // Collectors
+  private final SpringComponentsCollector springComponentsCollector;
+  private final PropertiesCollector propertiesCollector;
+
+  /**
+   * Get Spring components information in parallel.
+   *
+   * @return SpringComponentsInfo containing status of GMS, MAE Consumer, and MCE Consumer
+   */
+  public SpringComponentsInfo getSpringComponentsInfo() {
+    return springComponentsCollector.collect(executorService);
+  }
+
+  /**
+   * Get all system properties with detailed metadata.
+   *
+   * <p>Returns comprehensive property information including:
+   *
+   * <ul>
+   *   <li>Individual property details with sources and resolution
+   *   <li>Property source metadata
+   *   <li>Filtering and redaction statistics
+   * </ul>
+   *
+   * @return SystemPropertiesInfo with detailed property metadata
+   */
+  public SystemPropertiesInfo getSystemPropertiesInfo() {
+    return propertiesCollector.collect();
+  }
+
+  /**
+   * Get only configuration properties as a simple map (for backward compatibility).
+   *
+   * <p>This method provides a simplified view of system properties without metadata, suitable for
+   * legacy integrations or simple configuration debugging.
+   *
+   * @return Map of property keys to resolved values
+   */
+  public Map<String, Object> getPropertiesAsMap() {
+    return propertiesCollector.getPropertiesAsMap();
+  }
+
+  /**
+   * Get complete system information - spring components and properties in parallel.
+   *
+   * <p>This is the primary method for retrieving comprehensive system information. It executes both
+   * component and property collection in parallel for optimal performance.
+   *
+   * @return SystemInfoResponse containing both component and property information
+   * @throws SystemInfoException if collection fails or times out
+   */
+  public SystemInfoResponse getSystemInfo() {
+    List<CompletableFuture<?>> futures =
+        Arrays.asList(
+            CompletableFuture.supplyAsync(this::getSpringComponentsInfo, executorService),
+            CompletableFuture.supplyAsync(this::getSystemPropertiesInfo, executorService));
+
+    try {
+      CompletableFuture<Void> allFutures =
+          CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
+      allFutures.get(30, TimeUnit.SECONDS);
+
+      return SystemInfoResponse.builder()
+          .springComponents((SpringComponentsInfo) futures.get(0).getNow(null))
+          .properties((SystemPropertiesInfo) futures.get(1).getNow(null))
+          .build();
+    } catch (Exception e) {
+      log.error("Error collecting system info", e);
+      throw new SystemInfoException("Failed to collect system information", e);
+    }
+  }
+
+  @PreDestroy
+  public void shutdown() {
+    executorService.shutdown();
+    try {
+      if (!executorService.awaitTermination(5, TimeUnit.SECONDS)) {
+        executorService.shutdownNow();
```

Comment:
I kept the executor service from the original PR although I don't need to it yet. Also I'm not sure if (or why) we need the executor service when there are async methods available (like HttpClient's)